tests/adapter/dbt/tests/adapter/materialized_views/test_materialized_views.py

```diff
@@ -74,3 +81,50 @@ def test_updated_base_table_data_only_shows_in_materialized_view_after_rerun(sel
             self.get_records(project, self.materialized_view)
             == self.starting_records + self.inserted_records
         )
+
+
+class MaterializedViewTestsSkipConfigChangeBase(MaterializedViewTestsBase):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"models": {"on_configuration_change": "skip"}}
+
+    def test_on_configuration_change_skips_with_update(self, project):
+        results = run_dbt(
+            ["run", "--models", self.materialized_view, "--vars", "quoting: {identifier: True}"]
+        )
+        assert results.results[0].adapter_response["rows_affected"] == 0
+        results.results[0].node.config.on_configuration_change == "skip"
+        assert results.results[0].status == RunStatus.Success
+
+
+class TestMaterializedViewSkipTestsBase(MaterializedViewTestsSkipConfigChangeBase):
+    @pytest.mark.skip("This currently fails since we're mocking with a traditional view")
+    def test_updated_base_table_data_only_shows_in_materialized_view_after_rerun(self, project):
+        pass
+
+    def test_on_configuration_change_skips_with_update(self, project):
+        super().test_on_configuration_change_skips_with_update(self)
+
+
+class MaterializedViewTestsFailConfigChangeBase(MaterializedViewTestsBase):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"models": {"on_configuration_change": "fail"}}
+
+    def test_on_configuration_change_fails_with_update(self, project):
+        results = run_dbt(
+            ["run", "--models", self.materialized_view, "--vars", "quoting: {identifier: True}"],
+            expect_pass=False,
+        )
+        # breakpoint()
+        results.results[0].node.config.on_configuration_change == "fail"
+        assert results.results[0].status == RunStatus.Error
+
+
+class TestMaterializedViewFailTestsBase(MaterializedViewTestsFailConfigChangeBase):
```

Comment:
![Screenshot 2023-04-17 at 3 23 23 PM](https://user-images.githubusercontent.com/91097623/232601858-9a884649-0055-40f9-a746-e209c4d92ae7.png)

not sure if the exception we are wanting to throw will kick off as we might be hitting this one first